### PR TITLE
Remove unsupported fields from PutMappingRequest

### LINKED ISSUE #62: [BUG] Bad request response when calling the PutMapping api
**What is the bug?**
After calling the index api to make a put mapping request the following response is received:

```
org.opensearch.client.ResponseException: method [PUT], host [localhost:9200], URI [/memory/_mapping], status line [HTTP/1.1 400 Bad Request]
{"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"Root mapping definition has unsupported parameters:  [source_field : {enabled=true}]"}],"type":"mapper_parsing_exception","reason":"Root mapping definition has unsupported parameters:  [source_field : {enabled=true}]"},"status":400}
```

**How can one reproduce the bug?**
After creating an opensearch client, create a 'PutMappingRequest object and fill it out as below:

```
      PutMappingRequest.Builder mappingsRequestBuilder = new PutMappingRequest.Builder().index(indexName);
        mappingsRequestBuilder = mappingsRequestBuilder.sourceField(
                new SourceField.Builder()
                    .enabled(mappingsJson.getJsonObject("_source").getBoolean("enabled"))
                    .build());
        mappingsRequestBuilder = mappingsRequestBuilder.routingField(
                new RoutingField.Builder()
                    .required(mappingsJson.getJsonObject("_routing").getBoolean("required"))
                    .build());

      this.client.indices().putMapping(mappingsRequestBuilder.build());
```

**What is the expected behavior?**
Successful update of the mapping metadata

**What is your host/environment?**
 - OS:MacOS
 - Version 11.6

**Do you have any screenshots?**
n/a

**Do you have any additional context?**
I believe this error occurs because the fields are incorrect in the serialization of the PutMappingRequest. For example on the 'source' field the key is 'source_field' when it should be 'source'

e.g. https://github.com/opensearch-project/opensearch-java/blob/main/java-client/src/main/java/org/opensearch/client/opensearch/indices/PutMappingRequest.java#L467

I believe this is the case of many fields in that object.

Thanks

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
Finishing up PR #288 

### Issues Resolved
Closes #62 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — DISMISSED** [review]

File: java-client/src/main/java/org/opensearch/client/opensearch/indices/PutMappingRequest.java [hunk 1]
```diff
@@ -72,30 +65,9 @@
...
-	private final SourceField source;
-
 	@Nullable
 	private final Boolean allowNoIndices;
 
-	@Nullable
-	private final Boolean dateDetection;
-
-	@Nullable
-	private final DynamicMapping dynamic;
```
[the remark this answers is not in the document]
Comment (VachaShah) [reply java-client/src/main/java/org/opensearch/client/opensearch/indices/PutMappingRequest.java:93]: This is my bad! Thank you @reta for pointing out. Let me test this out and see if any other fields are supported as well. I will update the PR with the changes.

[the remark this answers is not in the document]
Comment (VachaShah) [reply java-client/src/main/java/org/opensearch/client/opensearch/indices/PutMappingRequest.java:93]: @reta I tried all the fields and all fields work except `runtime`. Sample code:

```
PutMappingRequest.Builder mappingsRequestBuilder = new PutMappingRequest.Builder().index("index")
            .source(
                new SourceField.Builder()
                    .enabled(true)
                    .build())
            .routing(
                new RoutingField.Builder()
                    .required(false)
                    .build())
            .dynamic(DynamicMapping.Strict)
            .meta("key", JsonData.of("key value"))
            .fieldNames(new FieldNamesField.Builder().enabled(false).build())
            .dateDetection(false)
            .dynamicDateFormats(new ArrayList<>())
            .dynamicTemplates(new ArrayList<>())
            .numericDetection(false)
            .runtime("runtime", new RuntimeField.Builder().type(RuntimeFieldType.Boolean).build());
client.indices().putMapping(mappingsRequestBuilder.build());
```

To which I got the below error:
```
Mapping response: org.opensearch.client.opensearch.indices.get_mapping.IndexMappingRecord@44040454
Exception in thread "main" org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [mapper_parsing_exception] Root mapping definition has unsupported parameters:  [runtime : {runtime={type=boolean}}]
        at org.opensearch.client.transport.rest_client.RestClientTransport.getHighLevelResponse(RestClientTransport.java:278)
        at org.opensearch.client.transport.rest_client.RestClientTransport.performRequest(RestClientTransport.java:144)
        at org.opensearch.client.opensearch.indices.OpenSearchIndicesClient.putMapping(OpenSearchIndicesClient.java:1148)
        at hello.OpenSearchValidateClient.main(OpenSearchValidateClient.java:234)
```

I think I am using `runtime` parameter correctly but LMK if thats not the case.

[the remark this answers is not in the document]
Comment (reta) [reply java-client/src/main/java/org/opensearch/client/opensearch/indices/PutMappingRequest.java:93]: Thanks @VachaShah !

> I think I am using runtime parameter correctly but LMK if thats not the case.

Correct, we don't support runtime fields (this is ES only feature)

[the remark this answers is not in the document]
Comment (VachaShah) [reply java-client/src/main/java/org/opensearch/client/opensearch/indices/PutMappingRequest.java:93]: Ah okay! I will update this PR accordingly. Thank you :)

**Review by reta — APPROVED** [review]

**Comment by dblock** [comment]
I think a good practice we could start following to catch bugs like the one @reta pointed out in the CR is to start adding working samples to https://github.com/opensearch-project/opensearch-java/tree/main/samples as part of these changes. Just an idea!

**Review by dblock — APPROVED** [review]

**Comment by VachaShah** [comment]
> I think a good practice we could start following to catch bugs like the one @reta pointed out in the CR is to start adding working samples to https://github.com/opensearch-project/opensearch-java/tree/main/samples as part of these changes. Just an idea!

Yup that makes sense! I have a working sample with which I tested. I will add it in an upcoming PR.

**Pull request merged**